src/util/TypeGuards.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+/**
+ * Helpers for checking different variants of common types.
+ */
+class TypeGuards extends null {
+  /**
+   * Verifies if the given command data is chat input command or not.
+   * @param {ApplicationCommandData} commandData The command data to check.
+   * @returns {boolean} True if it conforms to chat input command data, false otherwise.
+   */
+  static isChatInputCommandData(commandData) {
+    return commandData.type === 'CHAT_INPUT' || commandData.type === 1;
+  }
+
+  /**
+   * Verifies if the given command data is a context menu command or not.
+   * @param {ApplicationCommandData} commandData The command data to check.
+   * @returns {boolean} True if it conforms to chat input command data, false otherwise.
+   */
+  static isContextMenuCommandData(commandData) {
+    return (
+      commandData.type === 'MESSAGE' || commandData.type === 'USER' || commandData.type === 2 || commandData.type === 3
+    );
+  }
+
+  /**
+   * Verifies if the given command option data supports choices or not.
+   * @param {ApplicationCommandOptionData} commandOptionData The command option data to check.
+   * @returns {boolean} True if the option supports choices, false otherwise.
+   */
+  static optionDataSupportsChoices(commandOptionData) {
+    return (
```

Comment:
possibly something like
```js
const type = typeof x.type === 'number' ? x.type : Enum[x.type];
return [Enum.X_TYPE, Enum.Y_TYPE].includes(type);
```
would be better (for each function)

Reply:
Thank you for this, I went ahead an implemented it.